src/seb/interfaces/mteb_task.py

```diff
@@ -76,8 +68,12 @@ def get_descriptive_stats(self) -> DescriptiveDatasetStats:
 
     def evaluate(self, model: Encoder) -> TaskResult:
         split = self.mteb_task.description["eval_splits"][0]
-        task_model = MTEBTaskModel(model, self)
-        scores = self.mteb_task.evaluate(task_model, split=split)
+        # Infusing task into encode()
+        original_encode = model.encode
+        model.encode = partial(model.encode, task=self)
+        scores = self.mteb_task.evaluate(model, split=split)
+        # Resetting encode to original
+        model.encode = original_encode
```

Comment:
If an error is raised here would get us some odd problems. One solution is to make a context manager another is to:

```suggestion
        # Infusing task into encode()
        original_encode = model.encode
        model.encode = partial(model.encode, task=self)
        try:
          scores = self.mteb_task.evaluate(model, split=split)
          model.encode = original_encode # reset encode function
        except Exception as e:
          # this ensure that models encode function is always reset
          model.encode = original_encode
          raise(e)

```

Reply:
Ahhh smart, I haven't even thought of that, thanks Kenneth